tests/PHPStan/Analyser/data/template-default.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace TemplateDefault;
```

Comment:
What is realworld example where default type is helpful over default implied maximal type (the type after `of`)?

Reply:
See the discussion at https://github.com/phpstan/phpstan/issues/4801

Reply:
The issue did not answer my question. Default implied maximal type would solve the:

 > Consider this: If a value passed for $a is null, it's impossible to resolve TResult's type. Then default type should be used. Currently, the default type everywhere is mixed instead.

too and in much safer way without any extra coding needed as long as `of` type is specified.

Reply:
Sorry, I don't get it. Please show the example in https://github.com/phpstan/phpstan/issues/4801 and discuss it there. I plan to merge this PR very close to as it is.

Other languages also offer "default types for @template".